webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -583,7 +583,7 @@ class ClassicTechniqueWriter(basePath : String, parameterTypeService: ParameterT
   def reportingContext(methodCall: MethodCall, classParameterValue: String ) = {
     val component  = escapeCFEngineString(methodCall.component)
     val value = escapeCFEngineString(classParameterValue)
-    s"""_method_reporting_context("${component}", "${value}")"""
+    s"""_method_reporting_context("${component}", "${value}","${methodCall.id}")"""
```

Comment:
we need to duplicate the method to avoid breaking stuff during upgrade

Reply:
and we need the matching PR in ncf before merging this one

Reply:
Yes that will be a duplicated method. I need to validate the POC first.